### ISSUE TITLE
Ionic v4: Re-add compareWith property to ion-select

### ISSUE THREAD

**Issue by FdezRomero** [ionic-team/ionic#14598]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
cli packages: (/usr/local/lib/node_modules)

   @ionic/cli-utils  : 2.0.0-rc.6
   ionic (Ionic CLI) : 4.0.0-rc.6

local packages:

   @angular-devkit/core       : 0.6.8
   @angular-devkit/schematics : 0.6.8
   @angular/cli               : 6.0.8
   @ionic/schematics-angular  : 1.0.0-rc.6
   Ionic Framework            : @ionic/angular 4.0.0-alpha.7

System:

   NodeJS : v8.11.1
   npm    : 6.1.0
   OS     : macOS High Sierra
```

**Describe the Bug**
`ion-select` is missing the `compareWith` Angular property in ionic-angular v4, previously working in v3 with PR #11965.

**Steps to Reproduce**
1. Add an `<ion-select>` element with `[compareWith]="compareFunction"`.
2. The following error is thrown in the console:
```
Can't bind to 'compareWith' since it isn't a known property of 'ion-select'.
1. If 'ion-select' is an Angular component and it has 'compareWith' input, then verify that it is part of this module.
2. If 'ion-select' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
```

**Related Code**
If you are able to illustrate the bug with an example, please provide a sample application via an online code collaborator such as [StackBlitz](https://stackblitz.com), or [GitHub](https://github.com).

**Expected Behavior**
To be able to bind to `compareWith` in an ion-select.

**Additional Context**
[Angular docs for compareWith](https://angular.io/api/forms/SelectControlValueAccessor#caveat-option-selection)

**Comment by izio38** [comment]
Just go over this file : https://github.com/ionic-team/ionic/blob/master/core/src/components/select/select.tsx

and set: 

```
  @Prop()
compareWith(fn: (o1: any, o2: any) => boolean) {
  if (typeof fn !== 'function') {
    throw new Error(`compareWith must be a function, but received ${JSON.stringify(fn)}`);
  }
  this._compareWith = fn;
}
```

Then build again the core with the `npm run build.dev.core` command and you're done! 👍

**Comment by matheusdavidson** [comment]
same problem, need compareWith to work with objects!

**Comment by bkarv** [comment]
Has there been any more work on this? I too need compareWith to work with objects. @FdezRomero are you saying izio38 solution won't work? If not did you find a work around, do you mind sharing it?

**Comment by bkarv** [comment]
Thanks for the prompt response. I looked at Angular Material, did you mean you gave up on Ionic all together or use import Angular Material's form components into your Ionic app?

**Comment by FdezRomero** [comment]
@bkarv Not disappointing at all. That's what betas are made for! Finding things that are missing or not working as expected and reporting them 😄

I didn't drop Ionic at all, I just replaced inputs and selects for the ones in Angular Material, since the drop-down select style suited by app better anyway.

**Comment by bkarv** [comment]
@FdezRomero sorry misunderstood! Ok well fingers crossed a solution is found.

**Comment by bkarv** [comment]
@FdezRomero @izio38 @matheusdavidson 

I needed compareWith because had an array of objects as my select options eg {id: string, name: string}[]

After digging around I found this post where we can used trackBy instead 

https://github.com/ionic-team/ionic/pull/6627 

Tried it in Ionic 4 -beta 8 and it works like a charm! If anything trackBy feature is better than compareWith as I don't have to define the function. Here is a working example below:

Controller
```
  name: {id: string, name: string} = null; // question 1
  list: {id: string, name: string}[] = [{id: '1', name: 'John'}, {id: '2', name:'Peter'}, {id: '3', name: 'David'}];
```
Html
```
<ion-select  trackBy="id"  placeholder="Choose Name" [(ngModel)]="name" >
          <ion-select-option *ngFor="let item of list"  class="word-wrap" [value]="item">{{item.name}} 
          </ion-select-option>
</ion-select>
```

**Comment by kirliam** [comment]
Same problem! And the trackBy suggested by @bkarv is not working on beta15.

**Comment by patryk-fuhrman** [comment]
Any updates in this matter? trackBy is not working on rc3 too.

**Comment by jkeczan** [comment]
+1 on this. I needed this consistently as I am working on an application with about 30 selects

**Comment by zakton5** [comment]
I'm hoping to create a PR for this fairly soon. Stay tuned!

**Comment by zakton5** [comment]
Put in a PR for this finally.

**Comment by remisture** [comment]
When will this be merged?

**Comment by liamdebeasi** [comment]
Hi everyone,

This has been fixed by #17358 and will be in the next release of Ionic. Thanks for your patience!
